packages/web3/src/v3/SpaceDapp.ts

```diff
@@ -900,11 +900,12 @@ export class SpaceDapp implements ISpaceDapp {
                 }
             }
 
-            const entitlements = await this.getChannelEntitlementsForPermission(
+            const entitlements = await this.getChannelEntitlementsForPermissionUncached(
```

Comment:
To clarify, the "Uncached" above was referring to the entitlementEvaluationCache (stores is or is not entitled), not the entitlementCache (stores entitlements.) The current implementation exactly mirrors what's written in go entitlement evaluation, which you can find in `auth_impl.go`.

I'm not against the change, but can I ask why? And also can you please propagate it there too? I think it's easier to reason about the system if our caching behavior is the same everywhere.

You may also want to reach out to @erikolsson and see if/how he is caching on the swift client.

Reply:
Agree this should mirror the go implementation, otherwise we will get inconsistent UX when doing various permission updates. Although, it's also easy to clear the local client cache if you just refresh the page, which we're doing during development.

@ttillberg what were you experiencing that caused you to change this? I'm sure its related to permission overrides - probably you're changing permissions and appears its not updating correctly?

FYI in the roles panel, when updating permissions, we flash a toast to the user that any permissions changes may take up to 15 minutes to take effect

@clemire do you know if 15 mins cache for positive entitlement result is still applied during local development (in both client and server code)?

Reply:
I see in SpaceDapp it adjusts to 5 seconds for local development

Reply:
Local dev is configured to a shorter cache ttl for ease of manual and automated testing. We use the 5s time in CI as well and have a few 2s (negative) / 5s (positive) waits in e2e test cases to compensate for caching when entitlements change.

Reply:
thanks so much for the clarification @clemire - this was a misunderstanding of mine. Indeed the issue I'm experience in tests is related to the cache TTL, just adding some wait before the contract calls fixes this. I'm wondering if we could potentially look at passing caching options in the dApp constructor for more flexibility?

Reply:
@ttillberg for the purposes of userops tests, we could add cache parameters to Dapp - those tests don't interact w/ river nodes. But for e2e tests, you'd actually have to run the server code w/ different cache times as well. Correct me if I'm wrong @clemire

Reply:
Yes, in order to reduce wait times for your test case, if you're adding an event to a river node in your logical flow, you'd also need to configure the node's cache ttls. But if 5s is too long, we *can* decrease it!

Reply:
5s should be noticeable if you'd like to manually test caching behavior in the UI, but not too long to wait when you need it to update.